src/main/java/com/google/sps/servlets/SignUpServlet.java

```diff
@@ -33,11 +33,7 @@ private boolean checkIfFieldExists(String field, String value, Datastore datasto
 			 .setFilter(PropertyFilter.eq(field, value))
 			 .build();
 		 QueryResults<Entity> entities = datastore.run(fieldQuery);
-		 boolean fieldExists = false;
-
-		 while(entities.hasNext() && !fieldExists){
-			 fieldExists = true;
-		 }
+		 boolean fieldExists = entities.hasNext();
```

Comment:
You can return this directly if you're not expecting to add anything else in the future i.e.
`return entities.hasNext();`

One thing to double check, can entities be null at any point since you cannot call a method on a null variable.

Reply:
In the QueryResults<Entity> documentation it says that it's possible to get a DatastoreException upon calling .hasNext() or next() so I'll throw the exception in the function and change the return to hasNext(). Will also do the same for SignInServlet